File: AMF5/src/test/java/javaPlatform/GraphQLConversionTest.java
package javaPlatform;

import amf.apicontract.client.platform.AMFBaseUnitClient;
import amf.apicontract.client.platform.AMFDocumentResult;
import amf.core.client.platform.AMFResult;
import amf.graphql.client.platform.GraphQLConfiguration;
import amf.graphqlfederation.client.platform.GraphQLFederationConfiguration;
import amf.graphqlfederation.internal.spec.transformation.GraphQLFederationIntrospectionPipeline;
import org.junit.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.concurrent.ExecutionException;
import java.util.stream.Collectors;

import static org.junit.Assert.*;

public class GraphQLConversionTest {

    final String origin_federation = "file://src/test/resources/examples/federation-origin.graphql";
    final Path converted_no_federation = Paths.get("src/test/resources/expected/federation-converted.jsonld");

    @Test
    public void transformGraphQLFederationToGraphQL() throws ExecutionException, InterruptedException, IOException {
        final AMFBaseUnitClient graphQLFedClient = GraphQLFederationConfiguration.GraphQLFederation().baseUnitClient();
        final AMFBaseUnitClient graphQLClient = GraphQLConfiguration.GraphQL().baseUnitClient();

        final AMFDocumentResult parseResult = graphQLFedClient.parseDocument(origin_federation).get();

        assertTrue(parseResult.conforms());
        assertNotNull(parseResult.document());

        final AMFResult transformResult = graphQLFedClient.transform(parseResult.document(), GraphQLFederationIntrospectionPipeline.name());

        assertTrue(transformResult.conforms());
        assertNotNull(transformResult.baseUnit());

        final String renderedGraphQL = graphQLClient.render(transformResult.baseUnit(), "application/ld+json");

        assertNotNull(renderedGraphQL);

        final String goldenGraphQL = Files.readAllLines(converted_no_federation).stream().collect(Collectors.joining(System.lineSeparator()));

        assertEquals(normalized(goldenGraphQL), normalized(renderedGraphQL));
    }

    private String normalized(String s) {
        return s.replaceAll("(?s)\\s+", " ").trim();
    }
}

File: AMF5/src/test/scala/scalaPlatform/GraphQLConversionTest.scala
package scalaPlatform

import amf.apicontract.client.scala.AMFBaseUnitClient
import amf.graphql.client.scala.GraphQLConfiguration
import amf.graphqlfederation.client.scala.GraphQLFederationConfiguration
import amf.graphqlfederation.internal.spec.transformation.GraphQLFederationIntrospectionPipeline
import org.junit.runner.RunWith
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should.Matchers
import org.scalatestplus.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class GraphQLConversionTest extends AsyncFlatSpec with Matchers {

  private val origin_federation = "file://src/test/resources/examples/federation-origin.graphql"
  private val converted_no_federation = "src/test/resources/expected/federation-converted.jsonld"

  val graphQLFedClient: AMFBaseUnitClient =
    GraphQLFederationConfiguration.GraphQLFederation.baseUnitClient
  val graphQLClient: AMFBaseUnitClient = GraphQLConfiguration.GraphQL().baseUnitClient

  "AMF" should "convert GraphQLFederation to GraphQL" in {
    graphQLFedClient.parse(origin_federation) map { parseResult =>
      assert(parseResult.conforms)
      val transformResult = graphQLFedClient.transform(
        parseResult.baseUnit,
        GraphQLFederationIntrospectionPipeline.name
      )
      assert(transformResult.conforms)
      val renderedGraphQL = graphQLClient.render(transformResult.baseUnit, "application/ld+json")
      val goldenGraphQL = getStrFromFile(converted_no_federation)

      normalized(goldenGraphQL) shouldEqual(normalized(renderedGraphQL))
    }
  }

  private def getStrFromFile(path: String): String = {
    val source = scala.io.Source.fromFile(path)
    val read =
      try source.mkString
      finally source.close()
    read
  }

  def normalized(s: String): String = s.replaceAll("(?s)\\s+", " ").trim
}
